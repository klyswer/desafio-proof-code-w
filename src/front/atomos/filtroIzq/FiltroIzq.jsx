import React from 'react';
import PropTypes from 'prop-types';
import './style.css';


const FiltroIzq = ({listaElm=[],title}) => {
  return (
    <div className="contenedor_filtro">
      <h5 className="filtro_galeria">{title}</h5>
      {
        listaElm.length >0 && (
          listaElm.map((item)=>(
            <div key={item.id}>
              {item.opcion}
              <span>{` ${item.value}`}</span>
            </div>
          ))
        )
      }
    </div>
  );
};


FiltroIzq.propTypes = {
  listaElm: PropTypes.arrayOf(PropTypes.object).isRequired,
};


export default FiltroIzq;
