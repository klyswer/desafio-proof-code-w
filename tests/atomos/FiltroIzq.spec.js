import React from 'react';
/* eslint-disable no-undef */
import { shallow } from 'enzyme';
import FiltroIzq from '../../src/front/atomos/filtroIzq/FiltroIzq';

describe('<FiltroIzq />', () => {

  const listaFiltros=[{"id": '1krp',"opcion": "Marca1","value":"Datos"}];
  const titleFilter = "Filtro de prueba";
  const wrapper = shallow( <FiltroIzq listaElm={listaFiltros} title={titleFilter}  /> );


  test('Primer SnapShot: <FiltroIzq />', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Caso prueba 1: validando render de props title:string.', () => {
    const titleEsperado = wrapper.find('.filtro_galeria').text();
    expect(titleEsperado).toEqual(titleFilter);
  });

});
