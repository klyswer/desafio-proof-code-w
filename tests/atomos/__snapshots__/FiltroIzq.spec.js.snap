// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FiltroIzq /> Primer SnapShot: <FiltroIzq /> 1`] = `
<div
  className="contenedor_filtro"
>
  <h5
    className="filtro_galeria"
  >
    Filtro de prueba
  </h5>
  <div
    key="1krp"
  >
    Marca1
    <span>
       Datos
    </span>
  </div>
</div>
`;
